Micu/mark currently applicable line in the opening hours in venue description #289

micuintus opened the pull request.

jmastr commented:
![309_2](https://user-images.githubusercontent.com/11860038/68087609-e3dc8280-fe57-11e9-9747-654a03438f35.png)

@micuintus Don't know, if it is part of your patch, but shouldn't be the opening hours here in one line?

micuintus commented:
@jmastr Good catch!

> Don't know, if it is part of your patch,

No, this was already part of #237

> but shouldn't be the opening hours here in one line?

The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Subday to Sunday / Holdiday, but now the collapsing looks akward for that, which is why we removed it.

jmastr commented:
> The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Subday to Sunday / Holdiday, but now the collapsing looks akward for that, which is why we removed it.
We can leave it for now, but this solution isn't ideal either...

micuintus commented:
> > The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Sunday to Sunday / Holiday, but now the collapsing looks akward for that, which is why we removed it.
> 
> We can leave it for now, but this solution isn't ideal either...

Agreed, its not. But: It was not introduced within the scope of this ticket :D
On the other hand, one could argue that marking the open line mitigates the issue why we renamed  Sunday to Sunday / Holiday, a bit. But I'd say lets leave this for another ticket.

jmastr commented on src/OpeningHoursAlgorithms.h:
```diff
@@ -30,27 +30,30 @@
     if (from == until)
     {
-        return openingHours[from].toMap();
+        result = openingHours[from].toMap();
```
OpeningHoursAlgorithms.h:40: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

jmastr commented on src/OpeningHoursAlgorithms.h:
```diff
@@ -30,27 +30,30 @@
     {
-        QVariantMap result;
-
         const auto& fromDay  = openingHours[from]. toMap()["day"].toString();
```
OpeningHoursAlgorithms.h:44: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

jmastr commented on src/OpeningHoursAlgorithms.h:
```diff
@@ -30,27 +30,30 @@
-        QVariantMap result;
-
         const auto& fromDay  = openingHours[from]. toMap()["day"].toString();
         const auto& untilDay = openingHours[until].toMap()["day"].toString();
```
OpeningHoursAlgorithms.h:45: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

jmastr commented on src/OpeningHoursAlgorithms.h:
```diff
@@ -30,27 +30,30 @@
-
-        return result;
+        result["day"]      = fromDay + " - " + untilDay;
+        result["hours"]    = openingHours[from].toMap()["hours"];
```
OpeningHoursAlgorithms.h:48: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

jmastr commented on src/OpeningHoursAlgorithms.h:
```diff
@@ -30,27 +30,30 @@
+        result["hours"]    = openingHours[from].toMap()["hours"];
     }
+
+    result["current"]  = from <= todayIndex && todayIndex <= until;
```
OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const unsigned int' and 'const int'
OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const int' and 'const unsigned int'

jmastr commented on src/VenueModel.cpp:
```diff
@@ -282,15 +244,28 @@ QVariant VenueModel::data(const QModelIndex &index, int role) const
+    default:
+        return QStandardItemModel::data(index, role);
+    }
 }
```
VenueModel.cpp:270: warning: control may reach end of non-void function [-Wreturn-type]

jmastr commented:
Let's tackle that in a follow-up ticket, since the PR did not introduce the warning and the PRs is big enough already.
